hoot-core/src/main/cpp/hoot/core/io/OsmApiWriter.cpp

```diff
@@ -0,0 +1,449 @@
+/*
+ * This file is part of Hootenanny.
+ *
+ * Hootenanny is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * --------------------------------------------------------------------
+ *
+ * The following copyright notices are generated automatically. If you
+ * have a new notice to add, please use the format:
+ * " * @copyright Copyright ..."
+ * This will properly maintain the copyright information. DigitalGlobe
+ * copyrights will be updated automatically.
+ *
+ * @copyright Copyright (C) 2018 DigitalGlobe (http://www.digitalglobe.com/)
+ */
+
+#include "OsmApiWriter.h"
+
+//  Hootenanny
+#include <hoot/core/VersionDefines.h>
+#include <hoot/core/io/OsmApiChangeset.h>
+#include <hoot/core/io/OsmApiChangesetElement.h>
+#include <hoot/core/util/ConfigOptions.h>
+#include <hoot/core/util/FileUtils.h>
+#include <hoot/core/util/HootException.h>
+#include <hoot/core/util/Log.h>
+
+//  Qt
+#include <QXmlStreamReader>
+
+using namespace std;
+
+namespace hoot
+{
+
+OsmApiWriter::OsmApiWriter(const QUrl &url, const QString &changeset)
+  : _description(ConfigOptions().getChangesetDescription()),
+    _maxWriters(ConfigOptions().getChangesetApidbMaxWriters()),
+    _maxChangesetSize(ConfigOptions().getChangesetApidbMaxSize())
+{
+  _changesets.push_back(changeset);
+  if (isSupported(url))
```

Comment:
I think an error should be logged somewhere if the URL isn't supported. I may have missed the log call - but if I didn't, this might be a good place? Same for line 63.